### ISSUE TITLE
Scalability

### ISSUE THREAD

**Issue by leonard-gleyzer** [leonard-gleyzer/connex#1]
During the forward pass, connex uses the columns of an adjacency matrix as masks so that `jax.vmap` can work with neurons that have different numbers of inputs. However, this results in noticeably poor scaling, even with `vmap`.
